### PR TITLE
[DM-23942] Refactor JWT Authorizer

### PR DESCRIPTION
**Pull request by rra**
- Move more code into the Ticket, Session, and SessionStore classes
- Reduce duplicate code in test suite setup
- Remove unused configuration parameters
- Remove unnecessary complexity in how authorization checks are done
- Add tests for all the major code paths
- Refactor user-generated token issuance into a TokenStore class

### CODE REVIEW

File: src/jwt_authorizer/session.py [hunk 1]
```diff
@@ -167,6 +169,29 @@ def get_session(self, ticket: Ticket) -> Optional[Session]:
 
         return self._decrypt_session(ticket.secret, encrypted_session)
 
+    def store_session(
+        self, ticket: Ticket, session: Session, pipeline: Pipeline
+    ) -> None:
+        """Store an oath2_proxy session in the provided Redis pipeline.
```
Comment (cbanek) [inline src/jwt_authorizer/session.py]: oauth2_proxy

File: tests/app_auth_test.py [hunk 2]
```diff
@@ -309,3 +309,69 @@ def test_authnz_token_reissue(redis_client: redis.Redis) -> None:
...
+    now = time.time()
+    expected_exp = now + app.config["OAUTH2_JWT_EXP"] * 60
+    assert expected_exp - 5 <= decoded_token["exp"] <= expected_exp + 5
+    assert now - 5 <= decoded_token["iat"] <= now + 5
+
+    session_store = SessionStore("oauth2_proxy", session_secret, redis_client)
+    session = session_store.get_session(ticket)
+    assert session
+    assert session.token == new_token
+    assert session.user == "some-user@example.com"
```
Comment (cbanek) [inline tests/app_auth_test.py]: I honestly don't think I understand what this reissuance is, but the tests look good!

File: src/jwt_authorizer/app.py [hunk 3]
```diff
@@ -449,8 +452,7 @@ def _check_reissue_token(
             store_user_info=False,
             oauth2_proxy_ticket=ticket,
         )
-        oauth2_proxy_ticket_str = ticket.encode(cookie_name)
-    return encoded_token, oauth2_proxy_ticket_str
+    return encoded_token, ticket.encode(cookie_name) if ticket else ""
```
[the remark this answers is not in the document]
Comment (rra) [reply src/jwt_authorizer/app.py]: It's not. This code is dodgy and weird. I put off simplifying it for now and just preserved the existing behavior of returning the empty string in some odd code paths that I don't think happen in practice, but I am going to revisit it later and think harder about how we want to reissue tokens.

File: tests/authnz_test.py [hunk 4]
```diff
@@ -62,4 +62,7 @@ def test_capabilities_from_groups() -> None:
 
         admin_token = copy.deepcopy(token)
         admin_token["isMemberOf"].append({"name": "admin"})
-        assert capabilities_from_groups(admin_token) == {"exec:admin"}
+        assert capabilities_from_groups(admin_token) == {
```
[the remark this answers is not in the document]
Comment (rra) [reply tests/authnz_test.py]: No, just a change in the tests. I added more group mappings for the `exec:admin` scope when improving tests elsewhere.

**Comment by rra** [comment]
Thanks for the review!

**Pull request merged**